components/brave_wallet_ui/components/shared/add-custom-token-form/add-custom-token-form.tsx

```diff
@@ -0,0 +1,348 @@
+  useLib,
+  useTokenInfo
+} from '../../../common/hooks'
+
+interface Props {
+  contractAddress: string | undefined
+  onHideForm: () => void
+}
+
+export const AddCustomTokenForm = (props: Props) => {
+  const {
+    contractAddress,
+    onHideForm
+  } = props
+
+  // state
+  const [hasError, setHasError] = React.useState<boolean>(false)
+  const [showTokenIDRequired, setShowTokenIDRequired] = React.useState<boolean>(false)
+  const [showAdvancedFields, setShowAdvancedFields] = React.useState<boolean>(false)
+  const [showNetworkDropDown, setShowNetworkDropDown] = React.useState<boolean>(false)
+
+  // Form States
+  const [tokenName, setTokenName] = React.useState<string>('')
+  const [tokenID, setTokenID] = React.useState<string>('')
+  const [tokenSymbol, setTokenSymbol] = React.useState<string>('')
+  const [tokenContractAddress, setTokenContractAddress] = React.useState<string>(contractAddress || '')
+  const [tokenDecimals, setTokenDecimals] = React.useState<string>('')
+  const [coingeckoID, setCoingeckoID] = React.useState<string>('')
+  const [iconURL, setIconURL] = React.useState<string>('')
+  const [customAssetsNetwork, setCustomAssetsNetwork] = React.useState<BraveWallet.NetworkInfo>()
+
+  // redux
+  const userVisibleTokensInfo = useSelector(({ wallet }: { wallet: WalletState }) => wallet.userVisibleTokensInfo)
+  const fullTokenList = useSelector(({ wallet }: { wallet: WalletState }) => wallet.fullTokenList)
+  const selectedNetwork = useSelector(({ wallet }: { wallet: WalletState }) => wallet.selectedNetwork)
+  const addUserAssetError = useSelector(({ wallet }: { wallet: WalletState }) => wallet.addUserAssetError)
+
+  // custom hooks
+  const { getBlockchainTokenInfo } = useLib()
+  const {
+    onFindTokenInfoByContractAddress,
+    foundTokenInfoByContractAddress
+  } = useTokenInfo(getBlockchainTokenInfo, userVisibleTokensInfo, fullTokenList, selectedNetwork)
+  const {
+    onAddCustomAsset
+  } = useAssetManagement()
+
+  // Handle Form Input Changes
+  const handleTokenNameChanged = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (hasError) {
+      setHasError(false)
+    }
+    setTokenName(event.target.value)
+  }, [hasError])
+
+  const handleTokenIDChanged = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (hasError || showTokenIDRequired) {
+      setHasError(false)
+      setShowTokenIDRequired(false)
+    }
+    setTokenDecimals('0')
+    setTokenID(event.target.value)
+  }, [hasError, showTokenIDRequired])
+
+  const handleTokenSymbolChanged = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (hasError) {
+      setHasError(false)
+    }
+    setTokenSymbol(event.target.value)
+  }, [hasError])
+
+  const handleTokenAddressChanged = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (hasError) {
+      setHasError(false)
+    }
+    setTokenContractAddress(event.target.value)
+  }, [hasError])
```

Comment:
you are going to update state/re-render anyways, might just want to always clear the hasError state so you don't need deps in the array